eth_tester/backends/pyethereum/v20.py

```diff
@@ -51,31 +51,45 @@ def get_accounts(self):
     # Chain data
     #
     def get_block_by_number(self, block_number):
-        raise NotImplementedError("Must be implemented by subclasses")
+        return self.evm.chain.get_block_by_number(block_number)
 
     def get_block_by_hash(self, block_hash):
-        raise NotImplementedError("Must be implemented by subclasses")
+        return self.evm.chain.get_block(block_hash)
 
     def get_latest_block(self):
-        raise NotImplementedError("Must be implemented by subclasses")
+        return self.evm.chain.head
+
+    # NOTE: Added as a helper, might be more broadly useful
+    def get_state(self, block_hash=None):
+        if block_hash:
+            # Compute state at specific block
+            return self.evm.mk_poststate_of_blockhash(block_hash)
+        else:
+            # Return the most recent block if not specified
+            return self.evm.head_state
 
     def get_transaction_by_hash(self, transaction_hash):
-        raise NotImplementedError("Must be implemented by subclasses")
+        return self.evm.chain.get_transaction(transaction_hash)
 
     def get_transaction_receipt(self, txn_hash):
-        raise NotImplementedError("Must be implemented by subclasses")
+        transaction = self.get_transaction_by_hash(txn_hash)
+        state = self.get_state(block_number=transaction.block_hash)
+        return state.receipts
 
     #
     # Account state
     #
     def get_nonce(self, account, block_number=None):
-        raise NotImplementedError("Must be implemented by subclasses")
+        state = self.get_state(block_number)
```

Comment:
This looks like you're passing in a block number when the `get_state` function expects a hash?

Reply:
Haha, I totally did that.

Untested so far, obviously lol

Reply:
Updated